src/Controller/SentencesController.php

```diff
@@ -294,6 +305,7 @@ public function add_an_other_sentence()
         if ($userLevel < 0) {
             return;
         }
+        $this->update_user_last_contribution($userId);
```

Comment:
CakePHP offers a mechanism to perform some action on every creation or update of a model: the afterSave callback. Have a look at the `afterSave()` method in `UsersTable.php` or `SentencesTable.php`. You could call `update_user_last_contribution()` from there instead. The advantage is that:
* you call it *only once*, instead of calling it from every controller action potentially involved in sentence creation/update
* it is easier to write unit tests